vault/expiration.go

```diff
@@ -205,11 +230,31 @@ func (r *revocationJob) OnFailure(err error) {
 		return
 	}
 
+	if errIsUnrecoverable(err) {
+		r.m.logger.Trace("lease revocation returned unrecoverable error", "lease_id", r.leaseID, "err", err)
+
+		// TODO 1978 below here should be incorporated into markLeaseAsZombie
+		le, loadErr := r.m.loadEntry(r.nsCtx, r.leaseID)
+		if loadErr != nil {
+			r.m.logger.Warn("failed to mark lease as zombie - failed to load", "lease_id", r.leaseID, "err", loadErr)
+			return
+		}
+
+		r.m.markLeaseAsZombie(r.nsCtx, le, err)
```

Comment:
I see that the nil check is in the `markLeaseAsZombie` function, but I think it would be helpful to have the lease check in the caller as well, so you can log the leaseID for which the corresponding loaded lease was nil.

Reply:
yup!